src/image.rs

```diff
@@ -0,0 +1,104 @@
+use crate::config::ListenerConfig;
+use std::sync::{Arc, RwLock};
+use byteorder::{ByteOrder, LittleEndian};
+use image::{ImageBuffer, DynamicImage, RgbaImage};
+
+use rosrust;
+use rosrust_msg;
+
+
+fn read_img_msg(img_msg: rosrust_msg::sensor_msgs::Image) -> DynamicImage {
+    let bytes = convert_u8(img_msg.data, &img_msg.encoding);
+    if img_msg.encoding.contains("rgb") {
+        return DynamicImage::ImageRgb8(ImageBuffer::from_raw(
+                img_msg.width,
+                img_msg.height,
+                bytes).unwrap())
+    } else {
+        return DynamicImage::ImageLuma8(ImageBuffer::from_raw(
+                img_msg.width,
+                img_msg.height,
+                bytes).unwrap())
+    }
+}
+
+fn convert_u8(vec: Vec<u8>, encoding: &str) -> Vec<u8> {
+    let mut bytes: Vec<u8> = Vec::new();
+    if encoding.contains("8") {
+        return vec  // already u8 encoding, do nothing
+    } else if encoding.contains("32F") {
+        let mut vals: Vec<f32> = Vec::new();
+        let mut max_val = f32::MIN;
+        let mut min_val = f32::MAX;
+        for elem in vec.chunks(4) {
+            let val = LittleEndian::read_f32(&elem);
+            vals.push(val);
+            if val > max_val { max_val = val; }
+            if val < min_val { min_val = val; }
+        }
+        for val in vals {
+            bytes.push(((val - min_val) / (max_val - min_val) * u8::MAX as f32) as u8);
```

Comment:
It could be that this magic is something you have to know when working in the field, but I have no clue what is happening here

Reply:
its normalizing the values between 0 and 255. so what we get maximum contrast for luma.
I now found https://github.com/mazznoer/colorgrad-rs and https://github.com/Ogeon/palette.  I will check this evening if it can be used easily. 
The least i can do is move this to a separate function